h/views/groups.py

```diff
@@ -149,25 +144,43 @@ def read_unauthenticated(group, request):
     return {'group': group}
 
 
-@view_config(route_name='group_read_noslug', request_method='GET')
-def read_noslug(group, request):
-    _check_slug(group, request)
+@view_defaults(route_name='group_read',
```

Comment:
So, there are three different `group_read` cases here, each guarded by a set of predicates that determines whether they apply:

- User logged in, has permission to read annotations in the group: Redirect to search page if that feature is enabled, show legacy group overview page otherwise
- User logged in, does not have permissions to read annotations in the group: Show group join page
- User logged out: Show group join page. The user is invited to log in first, after which they are redirected back to the join page